src/api/udf/Histogram.h

```diff
@@ -112,38 +116,40 @@ class Hist : public BaseType {
     folly::Histogram<InputType> histogram_;
     std::string json_;
 
+#define SAVE_VALUE(value) \
+  if (TypeDetect<InputType>::tid() == TypeDetect<int8_t>::tid() ||        \
+      TypeDetect<InputType>::tid() == TypeDetect<int16_t>::tid() ||       \
+      TypeDetect<InputType>::tid() == TypeDetect<int32_t>::tid() ||       \
+      TypeDetect<InputType>::tid() == TypeDetect<int64_t>::tid()) {       \
+        json.Int64(value);                             \
+      }                                                \
+  else if (TypeDetect<InputType>::tid() == TypeDetect<float>::tid() ||    \
+      TypeDetect<InputType>::tid() == TypeDetect<double>::tid()) {        \
+        json.Double(value);                            \
+      }
```

Comment:
this is simpler and faster, :)

> if constexpr (std::is_floating_point<InputType>::value){
  json.Int64(value);
}else {
  json.double(value);
}